Update PaperTrail dependency

manuelmeurer opened the pull request:
I want to use this with PaperTrail v8. Just opening a PR to get Travis going for now. 😄

manuelmeurer commented:
Could someone review my changes? The tests are green now...

DimaSamodurov commented on Gemfile:
```diff
@@ -15,6 +15,6 @@ elsif ENV['RAILS'] == "4.1"
 else # Rails 4.2
-  gem 'globalize', '~> 5.0'
-  gem 'paper_trail', '4.0.0.beta2'
+  gem 'globalize', '5.1.0.beta2'
```
Do we really need to stick with particular (beta) version? This would lead to version incompatibility in many projects. Can we keep '~> 5.0', so that all 5.X versions would work?

manuelmeurer replied:
You're right, I just wanted to test if it works with the latest Globalize version. Unfortunately Globalize v5.0.1 is more than 1.5 years old...

DimaSamodurov commented on globalize-versioning.gemspec:
```diff
@@ -18,9 +18,9 @@ Gem::Specification.new do |s|
   s.add_dependency 'activerecord', '>= 3.2.0', '< 5'
   s.add_dependency 'activemodel', '>= 3.2.0', '< 5'
-  s.add_dependency 'globalize', '>= 3.0.4', '< 6'
+  s.add_dependency 'globalize', '5.1.0.beta2'
```
Can we keep previous version restriction?

manuelmeurer replied:
Yeah, I will change it.

manuelmeurer commented:
I removed the commit that updated the Travis config since it has nothing to do with updating the paper_trail dependency. 8fc053eafb2aec1a90e261e275a22bbb66e864e0 hasn't either, but without it the tests did not run for me at all...

manuelmeurer commented:
PaperTrail 8 requires Ruby >= 2.2.0.
Would it be an option to release a new major version of this gem that drops support for Ruby < 2.2?
[Support for these version has ended anyways.](https://www.ruby-lang.org/en/news/2017/04/01/support-of-ruby-2-1-has-ended/)

DimaSamodurov commented:
Bumping up the version of the gem looks like a natural way to add support of a new version of a dependency.

manuelmeurer commented:
Great, I'll update the dependencies then! I guess you'll want to do the version bump after merging, so it shouldn't be part of this PR, right?

manuelmeurer commented:
I had to remove support for Rails < v4.2 and Globalize < v5 as well since PaperTrail v8 don't support them anymore ([Rails v4.1 is not maintained anymore anyways](http://guides.rubyonrails.org/maintenance_policy.html))

manuelmeurer commented:
@DimaSamodurov Could you have another look? Thanks!

vestimir commented:
What's the status here – CI passes, can we merge this into master and release a new version?

ikanyu commented:
Any update on this PR? 🙂

manuelmeurer commented:
@DimaSamodurov Anything I can do to help get this merged?

manuelmeurer commented:
I updated the PaperTrail dependency to the latest version 9 and dropped Ruby 2.3 (because PaperTrail dropped support for it and because it's EOL).
@DimaSamodurov I'd still be very interested to get this merged! Happy to squash the commits.

sergey-alekseev commented:
@simi can you merge this PR?  

The latest released version of PaperTrail that meets the current dependency is [v4.2.0](https://github.com/paper-trail-gem/paper_trail/tree/v4.2.0) which has been released more than 2 years ago.

manuelmeurer commented on globalize-versioning.gemspec:
```diff
@@ -16,11 +16,12 @@ Gem::Specification.new do |s|
+  s.add_dependency 'activerecord', '~> 4.2'
+  s.add_dependency 'activemodel',  '~> 4.2'
+  s.add_dependency 'globalize',    '~> 5.0'
+  s.add_dependency 'paper_trail',  '~> 9.0'
```
What do you suggest? My goal is to be able to use the latest PaperTrail version.
If this gem should support multiple versions (e.g. `>= 5.0`), the tests would need to be run with each version.

sergey-alekseev replied:
@manuelmeurer ideally a new gem version should be released per every major PaperTrail version I think. In this case a gem owner at least may use https://github.com/globalize/globalize-versioning/pull/17/commits/2e64af75f2304f476816262027212568d4bf61a4 and https://github.com/globalize/globalize-versioning/pull/17/commits/b69de62eafbbacdac55e52d2c1825ea10d87045c to release 2 versions.

manuelmeurer replied:
Sure, I'm happy to make the needed changes. @DimaSamodurov @simi What do you think?

parndt commented:
sorry for missing this @manuelmeurer - I just merged another PR so this has a few conflicts

manuelmeurer commented:
@parndt Well, this PR is not necessary anymore since #21 upgraded the PaperTrail dependency already. 😄
